cli/tests/test_agent_discovery_multi_install.py

```
@@ -0,0 +1,407 @@
+    home = tmp_path / "home"
+    home.mkdir()
+    _write_pkg_json(
+        home / ".npm-global/lib/node_modules/@openai/codex/package.json",
+        "0.150.0",
+    )
+    # Simulate a stale Caskroom entry at 0.100.0 below the min.
+    caskroom_versions = {"/opt/homebrew/Caskroom/codex": ["0.100.0"]}
+    original_listdir = os.listdir
+    original_isdir = os.path.isdir
+
+    def fake_listdir(p):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `fake_listdir`

(ANN202)